docs/row-hasher.md

```diff
@@ -0,0 +1,109 @@
+# Row hash prefix recipe
+
+## Background
+
+Transactions are implemented in Fluo using conditional mutations.  Conditional
+mutations require server side processing on tservers.  If data is not spread
+evenly, it can cause some tservers to execute more conditional mutations than
+others.  These tservers doing more work can become a bottleneck.  Most real
+world data is not uniform and can cause this problem.
+
+Before the Fluo [Webindex example][1] started using this recipe it suffered
+from this problem.  The example was using reverse dns encoded URLs for row keys
+like `p:com.cnn/story1.html`.  This made certain portions of the table more
+popular, which in turn made some tservers do much more work.  This uneven
+distribution of work lead to lower throughput and uneven performance.  Using
+this recipe made those problems go away.
+
+## Solution
+
+This recipe provides code to help add a hash of the row as a prefix of the row.
+Using this recipe rows are structured like the following.
+
+```
+<prefix>:<fixed len row hash>:<user row>
+```
+
+The recipe also provides code the help generate split points and configure
+balancing of the prefix.
+
+## Example Use
+
+```java
+import io.fluo.api.data.Bytes;
+import io.fluo.recipes.common.Pirtos;
+import io.fluo.recipes.data.RowHasher;
+
+public class RowHasherExample {
+
+  private static final RowHasher PAGE_ROW_HASHER = new RowHasher("p");
+
+  public static RowHasher getPageRowHasher(){
+    return PAGE_ROW_HASHER;
+  }
+
+  public static void main(String[] args) {
+    RowHasher pageRowHasher = getPageRowHasher();
```

Comment:
why make it private and have a getter?  why not just do `final RowHasher pageRowHasher = new RowHasher("p")` in main()?